include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -9569,6 +9569,12 @@ def err_bounds_type_annotation_lost_checking : Error<
 
   def err_initializer_expected_for_ptr : Error<
     "automatic variable %0 with _Ptr type must have initializer">;
+
+  def err_initializer_not_null_terminated_for_nt_checked : Error<
+    "null terminator expected in _Nt_checked array initializer">;
+
+  def err_initializer_not_null_terminated_for_nt_checked_string_literal : Error<
+     "initializer-string for _Nt_checked char array is too long">;

```

Comment:
I think this could be a little confusing to a developer.   I would suggest "initializer string for _Nt_checked char array does not initialize last array element to null."

Reply:
David, this happens only if the string literal assigned to the character array is longer than the size specified in the declaration. The warning shown by clang is "initializer-string for char array is too long". I tried to stick to a similarly worded message for checked C. Please let me know if you still want the change you've suggested here.